cmd/traefik/traefik.go

```diff
@@ -360,7 +360,23 @@ func getHTTPChallengeHandler(acmeProviders []*acme.Provider, httpChallengeProvid
 
 func getDefaultsEntrypoints(staticConfiguration *static.Configuration) []string {
 	var defaultEntryPoints []string
+
+	// Determines if at least one EntryPoint is configured to be in the default set.
+	var userDefinedDefaults bool
+	for _, ep := range staticConfiguration.EntryPoints {
+		if ep.DefaultSet {
+			userDefinedDefaults = true
+			break
+		}
+	}
+
 	for name, cfg := range staticConfiguration.EntryPoints {
+		// Ignores EntryPoints that are marked as not part of default set,
+		// if at least one is configured to be in the default set.
+		if userDefinedDefaults && !cfg.DefaultSet {
+			continue
+		}
```

Comment:
Do you maybe want to invert this, so it by default is included in the default set?

I think the additional logic of "if none are defined as default, then all are" is a weird semantic.

Just a suggestion, but I think the config could be changed to e.g:

- `--entrypoints.<name>.skipAsDefault`
- `--entrypoints.<name>.skipDefaultSet`
- `--entrypoints.<name>.noDefaultSet`
- `--entrypoints.<name>.optOutAsDefault`

or something like that. Then have the setting default to `false`, as now, but then you don't need this extra logic

Reply:
> Indeed we choose to use an opt-in fashion because we thought it was better once you started to decide which EntryPoint should be in the default set, since it prevents having an EntryPoint in the default set only because you forgot to opt-out.

Yea that makes a lot more sense, especially since this feature was requested from a security perspective to begin with.

> We were worried that a `DefaultEntrypoints` option could be seen as an option to choose what EntryPoint should be set up automatically.

Brilliant foresight!

And yeah having the config inside the EntryPoint model is probably a lot more future-proof. Good call